<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<arg name="description" default="cbf_v2_codegen_test"/>
	
	<arg name="margin" default="30.0"/>
	<arg name="use_margin" default="false"/>
	<arg name="hwil" default="true"/>
	<arg name="readonly" default="false"/>

	<arg name="timegap" value="2.2"/>
	<arg name="s_min" value="15.0"/>
	<arg name="act_delay" value="0.2"/>
	<arg name="k_tg_0" value="0.1"/>
	<arg name="k_tg_1" value="0.05"/>
	<arg name="k_ca_0" value="1.0"/>
	<arg name="k_ca_1" value="0.25"/>
	<arg name="k_ca_2" value="1.0"/>
	<arg name="a_min_no_brake_check" value="-0.5"/>
		
	<param name="/margin" value="$(arg margin)"/>
	<param name="/use_margin" value="$(arg use_margin)"/>
	<param name="/hwil" value="$(arg hwil)"/>
	<param name="/readonly" value="$(arg readonly)"/>

	<param name="/timegap" value="$(arg timegap)"/>
	<param name="/s_min" value="$(arg s_min)"/>
	<param name="/act_delay" value="$(arg act_delay)"/>
	<param name="/k_tg_0" value="$(arg k_tg_0)"/>
	<param name="/k_tg_1" value="$(arg k_tg_1)"/>
	<param name="/k_ca_0" value="$(arg k_ca_0)"/>
	<param name="/k_ca_1" value="$(arg k_ca_1)"/>
	<param name="/k_ca_2" value="$(arg k_ca_2)"/>
	<param name="a_min_no_brake_check" value="$(arg a_min_no_brake_check)"/>

	<node pkg="can_to_ros" type ="subs_fs" name="subs_fs" output="screen" if="$(arg hwil)"/> 

	<node pkg="can_to_ros" type="lead_info" name="lead_info" output="screen" if="$(arg hwil)"/>

	<node pkg="can_to_ros" type="simple_mini_car_from_lead_distance" name="simple_mini_car_from_lead_distance" output="screen" if="$(arg hwil)"/>

	<!--node pkg="can_to_ros" type="cruise_start_safety_check" name="cruise_start_safety_check" output="screen" if="$(arg hwil)"/-->

	
	<node pkg="margin" type="margin_node" name="margin_node" output="screen" if="$(arg use_margin)">
	</node>	
 
	<node pkg="setpointreader" type="setpointreader_node" name="setpointreader_node" output="screen" required="true" >
		<remap from="/v_ref" to="/cmd_vel"/>
	</node>
	
	<node pkg="velocity_controller" type="velocity_controller_node" name="velocity_controller_node" output="screen">	
		<remap from="/vehicle/vel" to="/vel"/>	
		<remap from="/cmd_accel" to="/cmd_accel_pre"/>
	</node>
	
	<node pkg="cbf_v2" type="cbf_v2_node" name="cbf_v2_node" if="$(eval not readonly and hwil)">
	</node>
	
	<node pkg="cbf_v2" type="cbf_v2_node" name="cbf_v2_node_readonly" if="$(eval readonly and hwil)">
		<remap from="/cmd_accel" to="/cmd_accel_null"/>
	</node>
	
	
	
	<node pkg="can_to_ros" type ="rosbag_record.sh" name="bashscript2" output="screen" args="$(arg description) $(arg hwil)" /> 

	<node pkg="transfer_pkg" type="saveparam.py" name="saveparam" output="screen" args="$(arg description) $(arg hwil)">
	</node>
	

</launch>
